fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBDatabaseFactory.java

```diff
@@ -374,59 +206,12 @@ public synchronized void setDatacenterId(String datacenterId) {
         }
     }
 
-    /**
-     * Set whether additional run-loop profiling of the FDB client is enabled. This can be useful for debugging
-     * certain performance problems, but the profiling is also fairly heavy-weight, and so it is not generally
-     * recommended when performance is critical. This method should be set prior to the first {@link FDBDatabase}
-     * is returned by this factory, as otherwise, it will have no effect (i.e., run-loop profiling will not actually
-     * be enabled). Enabling run-loop profiling also places its output in the FDB client trace logs, so it only makes
-     * sense to call this method if one also calls {@link #setTrace(String, String)}.
-     *
-     * @param runLoopProfilingEnabled whether run-loop profiling should be enabled
-     * @see NetworkOptions#setEnableSlowTaskProfiling()
-     */
-    public synchronized void setRunLoopProfilingEnabled(boolean runLoopProfilingEnabled) {
-        if (inited) {
-            throw new RecordCoreException("run loop profiling can not be enabled as the client has already started");
-        }
-        this.runLoopProfilingEnabled = runLoopProfilingEnabled;
-    }
-
-    /**
-     * Set the number of threads per FDB client version. The default value is 1.
-     *
-     * @param threadsPerClientV the number of threads per client version. Cannot be less than 1
-     */
-    public static void setThreadsPerClientVersion(int threadsPerClientV) {
```

Comment:
I think you mean it'll have to be a public static method on the abstract class

Reply:
You are right, I missed that.  As per one of our prior discussions, I think it will have to set the value in the `Impl` class.  Definitely not idea, but you could mark it as `@Deprecated` and `API.Status.DEPRECATED` with a javadoc to switch to the `Impl` method, and then we can remove it from here in the next major release (when making a major release, we do attempt to clean up as much deprecation as we can)